### PR TITLE
Add extra publish test cases

### PR DESCRIPTION
**Pull request by dapplion**
Addresses an item of https://github.com/aragon/buidler-aragon/issues/77

Extends the current unique publish test to 4 tests:
- localhost
  - Repo doesn't exist; deploys new contract and repo
  - Repo exists, do a minor or patch with just content changing
  - Repo exists, do a major with new contract and content
- mainnet (_skipped, may be unstable_)
  - Repo exists, do a minor or patch with just content changing

### CODE REVIEW

**Comment by sohkai** [comment]
> mainnet (skipped, may be unstable)

Yeah... I wouldn't do this. I would assume that if it works on a local chain, it'll work on mainnet.

File: src/tasks/publish/index.ts [hunk 1]
```diff
@@ -249,3 +244,35 @@ async function _getChainId(bre: BuidlerRuntimeEnvironment): Promise<number> {
   const net = await provider.getNetwork()
   return net.chainId
 }
+
+/**
+ * Utility to parse appName from the aragon config
+ * @param appNameOrObj
+ * @param network
+ */
+function _parseAppName(
```
Comment (0xGabi) [inline src/tasks/publish/index.ts]: What you think of moving this to the `appName` util file?

Comment (dapplion) [reply src/tasks/publish/index.ts]: Let's do!

File: test/src/tasks/publish-task.test.ts [hunk 2]
```diff
@@ -5,82 +5,108 @@ import { AragonConfig } from '~/src/types'
...
+      )
+    else
+      assert.equal(
+        data.contractAddress,
+        repoState.contractAddress,
+        'Version contract address must be the same as last version'
+      )
+    assert.equal(
+      data.version,
+      semver.inc(repoState.version, bump),
```
Comment (0xGabi) [inline test/src/tasks/publish-task.test.ts]: Nice 💪

File: test/src/tasks/publish-task.test.ts [hunk 3]
```diff
@@ -91,9 +117,191 @@ describe(`Run publish task - ${testAppDir}`, function() {
       { path: 'contracts/TestContract.sol' },
       'artifact.json does not include expected data'
     )
+  }
+
+  /**
+   * Test util to fetch the root account
+   * @param bre
+   */
+  async function getRootAccount(
```
Comment (0xGabi) [inline test/src/tasks/publish-task.test.ts]: Maybe move this to test-helpers?

Comment (dapplion) [reply test/src/tasks/publish-task.test.ts]: Fixed with https://github.com/aragon/buidler-aragon/pull/89/commits/6140156bede94b83e63cc03532839f7291e0712b

File: src/tasks/publish/index.ts (hunk 1 above)
Comment (dapplion) [reply src/tasks/publish/index.ts]: After looking at this, this a utility specific to the publish task need to read the aragon config. So I kept the utility there but change the name to be less confusing. 

Fixed with https://github.com/aragon/buidler-aragon/pull/89/commits/7ccebcb274cb752e041356316c53440d791410f0

File: test/src/tasks/publish-task.test.ts [hunk 4]
```diff
@@ -91,9 +118,181 @@ describe(`Run publish task - ${testAppDir}`, function() {
...
+    })
+
+    after('Stop ganache instance', function() {
+      if (closeGanache) closeGanache()
+    })
   })
 
-  after('Stop ganache instance', function() {
-    if (closeGanache) closeGanache()
+  describe.skip('On the mainnet network', function() {
```
Comment (0xGabi) [inline test/src/tasks/publish-task.test.ts]: Should we remove it or there is sections that makes sense to stay?

Comment (dapplion) [reply test/src/tasks/publish-task.test.ts]: Let's keep them and decide latter

**Pull request merged**